app/src/main/java/com/example/defaultstyles/CustomCircleView.kt

```diff
@@ -15,12 +15,18 @@ private const val DEFAULT_HAS_BORDER = false
 class CustomCircleView : View {
 
     constructor(context: Context) : this(context, null)
-    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, DEFAULT_STYLE_ATTR)
     constructor(
         context: Context,
         attrs: AttributeSet?,
         defStyleAttr: Int
-    ) : super(context, attrs, defStyleAttr) {
+    ) : this(context, attrs, defStyleAttr, DEF_STYLE_RES)
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int,
+        defStyleRes: Int
+    ) : super(context, attrs, defStyleAttr, defStyleRes) {
```

Comment:
This one is only available from v21 I think, but same deal - without this, only the custom property (`CustomCircleView_hasBorder`) would be respected from the default style _res_ (the case where a default style attr is not specified).